Balancer-Bots/Large-Internal-Balance-Deposit-Withdrawal/src/agent.spec.ts

```diff
@@ -0,0 +1,206 @@
+import { BigNumber } from "ethers";
+import { Interface, formatEther } from "ethers/lib/utils";
+import { FindingType, FindingSeverity, Finding, HandleBlock, Network, ethers } from "forta-agent";
+import { createAddress, MockEthersProvider, TestBlockEvent } from "forta-agent-tools/lib/tests";
+import { provideHandleBlock } from "./agent";
+import BalanceFetcher from "./balance.fetcher";
+import { TOKEN_ABI, EVENT } from "./constants";
+import { AgentConfig, NetworkData } from "./utils";
+import { NetworkManager } from "forta-agent-tools";
+
+const VAULT_IFACE = new Interface(EVENT);
+const IRRELEVANT_IFACE = new ethers.utils.Interface(["event Event()"]);
+const VAULT_ADDRESS = createAddress("0x1");
+
+const getInternalBalanceChangeLog = (
+  emitter: string,
+  block: number,
+  user: string,
+  token: string,
+  delta: ethers.BigNumberish
+): ethers.providers.Log => {
+  return {
+    address: emitter,
+    blockNumber: block,
+    ...VAULT_IFACE.encodeEventLog(VAULT_IFACE.getEvent("InternalBalanceChanged"), [
+      user,
+      token,
+      ethers.BigNumber.from(delta),
+    ]),
+  } as ethers.providers.Log;
+};
+
+const getIrrelevantEvent = (emitter: string, block: number): ethers.providers.Log => {
+  return {
+    address: emitter,
+    blockNumber: block,
+    ...IRRELEVANT_IFACE.encodeEventLog(IRRELEVANT_IFACE.getEvent("Event"), []),
+  } as ethers.providers.Log;
+};
+
+const createFinding = (user: string, token: string, delta: ethers.BigNumberish) => {
+  let action, alertId;
+
+  BigNumber.from(delta).isNegative()
+    ? ((action = "withdrawal"), (alertId = "BAL-2-1"))
+    : ((action = "deposit"), (alertId = "BAL-2-2"));
+
+  return Finding.fromObject({
+    name: `Large internal balance ${action}`,
+    description: `InternalBalanceChanged event detected with large ${action}`,
+    alertId,
+    protocol: "Balancer",
+    severity: FindingSeverity.Info,
+    type: FindingType.Info,
+    metadata: {
+      user: user.toLowerCase(),
+      token: token.toLowerCase(),
+      delta: formatEther(delta.toString()),
```

Comment:
```suggestion
      delta: delta.toString(),
```
Related to a later comment.